portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -40,52 +42,69 @@
-      return;
-    }
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(numComments));
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      List<Entity> results = datastore.prepare(query).asList(FetchOptions.Builder.withLimit(numComments));
 
-    ArrayList<Comment> comments = new ArrayList<>();
-    for (Entity entity : results) {
-      long id = entity.getKey().getId();
-      String text = (String) entity.getProperty("comment");
-      comments.add(new Comment(id, text));
+      ArrayList<Comment> comments = new ArrayList<>();
+      for (Entity entity : results) {
+        long id = entity.getKey().getId();
+        String text = (String) entity.getProperty("comment");
+        String user = (String) entity.getProperty("user");
+        comments.add(new Comment(id, text, user));
+      }
+      String logoutUrl = userService.createLogoutURL("/#comments");
+      commentsJson.put("comments", convertToJson(comments));
+      commentsJson.put("loggedin", true);
+      commentsJson.put("user", userService.getCurrentUser().getEmail());
+      commentsJson.put("url", logoutUrl);
+    } else {
+      commentsJson.put("loggedin", false);
+      String loginUrl = userService.createLoginURL("/#comments");
+      commentsJson.put("url", loginUrl);
     }
-
     response.setContentType("text/html;");
-    response.getWriter().println(convertToJson(comments));
+    response.getWriter().println(commentsJson);
   }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get the input from the form.
-    String comment = request.getParameter("text-input");
-    long timestamp = System.currentTimeMillis();
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
```

should you return an error (403) if the user is logged out?